packages/medusa/src/repositories/money-amount.ts

```diff
@@ -27,20 +27,36 @@ export class MoneyAmountRepository extends Repository<MoneyAmount> {
     variantId: string,
     prices: Price[]
   ): Promise<MoneyAmount[]> {
-    const pricesNotInPricesPayload = await this.createQueryBuilder()
+    const existingPricesInPayload = await this.createQueryBuilder()
```

Comment:
suggestion(non-blocking): I played around with this a bit with the thought of running one query and this as a subquery. The following seems to be working as well (though I haven't tested it with 100000 prices in the request): 
```
const existingPricesIdsQuery = this.createQueryBuilder()
      .select("id")
      .where({
        variant_id: variantId,
        price_list_id: IsNull(),
      })
      .andWhere(
        new Brackets((qb) => {
          prices.forEach((price) => {
            qb.orWhere(
              new Brackets((qb) => {
                qb.where({
                  currency_code: price.currency_code,
                  region_id: IsNull(),
                }).orWhere({ region_id: price.region_id })
              })
            )
          })
        })
      )

    const obsoletePrices = await this.createQueryBuilder("ma")
      .where({
        variant_id: variantId,
        price_list_id: IsNull(),
      })
      .andWhere(`ma.id NOT IN (${existingPricesIdsQuery.getQuery()})`)
      .setParameters(existingPricesIdsQuery.getParameters())
      .getMany()

    return obsoletePrices
```